[IV-23-24] Objetivo 2

Morad02 opened the pull request:
# Sobre la estructura del repositorio
* [x] ¿He seguido las mejores prácticas en el nombre de las clases y ficheros y
   disposición de los mismos?
* [x] ¿He mirado tanto reglas generales (para cualquier repo) como reglas
específicas para el lenguaje con el que se está trabajando?

# Sobre el planteamiento
* [x] ¿Se han planteado una serie de issues, asignados a las historias de
usuario pertinentes, que clarifiquen qué es lo que necesitan los usuarios e
identifiquen las diferentes partes del problema?
* [x] ¿Los issues representan un problema, y no una tarea?

# Sobre el análisis del problema
* [x] ¿Se ha documentado qué análisis se ha hecho sobre el dominio para decir lo
   que se ha creado?
* [x] ¿Se ha documentado por qué se ha elegido que lo creado sea un objeto valor,
   una entidad o un agregado?
* [x] ¿He propuesto un producto mínimamente viable, que en muchos casos será un solo
   objeto valor que no dependa de ningún otro (y que sea la base de muchos
   otros)?.

# Sobre el código
* [x] ¿Todos los mensajes de commit explican el cambio, y no se
      limitan a repetir el nombre del fichero que se ha cambiado?
* [x] ¿Los mensajes de commit siguen el formato estándar y buenas prácticas?
* [x] ¿Se ha hecho una revisión real del código para comprobar que todos
   los atributos y funciones creadas están respaldadas por una HU?
* [x] ¿Todos los issues creados están relacionados con a una HU?
* [x] ¿Ha asignado el propietario/la propietaria del repositorio todos los
issues planteados al *milestone* correspondiente?
* [x] ¿Todos los cambios en el código están asignados a un issue al que se
      referencia en un commit?
* [x] ¿Se ha asignado al mismo milestone el PR que se ha hecho?
* [x] ¿Es el milestone sobre el que estoy trabajando un producto
   mínimamente viable? ¿O tengo que solicitar al product manager que precise de
   qué producto se trata?
* [x] ¿Se ha comprobado que el código entregado sea sintácticamente
correcto?

Morad02 committed:
Lenguaje de programación decidido closses #5

Se ha debatido en #5. Queda plasmado en iv.yaml

Morad02 committed:
Ya se ha decidido el lenguaje closes #5

JJ commented on src/Sugerir.ts:
```diff
@@ -0,0 +1,27 @@
+import {Entrenador} from "./Entrenador";
+import { Atleta } from "./Atleta";
+
+class Sugerir
```
Una clase nunca es un verbo, siempre es un sustantivo.

JJ commented on src/Entrenador.ts:
```diff
@@ -0,0 +1,36 @@
+    private _nivelCompromisos: Array<Compromiso>;
+    private _modalidadesEntreno: Array<ModalidadEntreno>;
+    private _disponibilidadHorarioSemanal : Array<TramoHorario>;
+    private _preferenciasContacto : Array<TipoContacto>;
```
¿Esto sirve a la lógica de negocio?

JJ commented on src/Sugerir.ts:
```diff
@@ -0,0 +1,27 @@
+class Sugerir
+{
+    private _entrenadores: Array<[Entrenador,Array<Atleta>]>; //Array de tuplas. Tupla de entrenador y atletas sugeridos 
+    private _atletas: Array<[Atleta,Array<Entrenador>]>; //Array de tuplas. Tupla de atleta y entrenadores sugeridos
```
Si ya están aquí los datos, ¿para qué quieres construir una clase?

JJ reviewed (changes requested):
Solo he comentado un issue, pero supongo que habrá otros que tengan el mismo problema. Sin issues bien planteados (es decir, que cada uno salga de una HU y planteen unp roblema) esto al final es un ejercicio de poner nombre a las cosas.
La modelización del problema debe incluir entender cómo hay que representar la información para que se puedan hacer comparaciones vectoriales; esto lo hemos hablado varias veces @MarioGuisado . ¿Realmente crees que con lo hecho se hace un modelo adecuado del problema? Si tuvieras que ponerte ahora mismo a programar la lógica de negocio, ¿cómo lo harías con lo que has aceptado?

Morad02 committed:
Establecido el .gitignore

Teniendo en cuenta el lenguaje a usar. Debatido en #5

Morad02 committed:
Solucionado #11 closes #11

Usando map. Permitirá un filtrado más eficiente o clustiring

Morad02 committed:
Solucionado #12 closes #12

Usando map. Permitirá un filtrado más eficiente o clustiring

Morad02 committed:
Solucionado #13 closes #13

Usando map. Permitirá un filtrado más eficiente o clustiring

Morad02 committed:
Solucionado #28 closes #28

Se ha usado un map para definir los tramos. Donde las claves son los dias de la semana y los valores son el vector de tramos horarios

Morad02 committed:
Solucionado #14 closes #14

Se ha creado una clase para gestionar y consultar el horario semanal.

Morad02 committed:
Solucionado #15 closes #15

Usando map. Permitirá un filtrado más eficiente o clustiring

Morad02 committed:
Solucionado #16 closes #16

Usando map. Permitirá un filtrado más eficiente o clustiring

Morad02 committed:
Solucionado #10 closes #10

Usando un number ya que solo es relevando el año

Morad02 committed:
Solucionado #7 closes #7

Usando una clase para modelar este objeto valor

Morad02 committed:
Solucionado #18 closes #18

Se ha solucionado usando una tupla de dos números. El primero es la edad mínima y el otro es la edad máxima

Morad02 committed:
Solucionado #8 closes #8

Se ha creado una clase para modelar el objeto-valor Entrenador

Morad02 committed:
Solucionado #29 closes #29

Creaod una clase entidad para gestionar el valor objeto atleta

Morad02 committed:
Solucionado #30 closes #30

Usando una clase entidad para gestionar el objeto valor entrenador

Morad02 committed:
Corregido nombre de clase Atleta

De Atelta a Atleta

MarioGuisado reviewed (approved):
# Lista de comprobación del propietario del repositorio
* [x] ¿Los issues que ha planteado la persona son efectivamente problemas y
evidencian comprensión de las historias de usuario? Si no es así, es mejor que
se le transmita antes de que empiece a crear cualquier código erróneo.
* [x] ¿Has comprobado que efectivamente el compañero/a ha seguido *su* lista de
      comprobación?
* [x] ¿El código creado es una modelización válida de las historias de usuario
      creadas por ti, y te sirve como base para programar la lógica de negocio?
* [x] ¿El código creado sigue las mejores prácticas para el lenguaje de
      programación que hayáis consensuado, tanto en identificadores, como
      nombres de ficheros, como disposición de los diferentes ficheros en el
      repositorio? Por ejemplo, ¿todos los commits se refieren a un issue y
      aportan código para resolver el problema planteado en el mismo?
* [x] ¿He comprobado que el código compile, o que quien lo ha programado ha lo
ha hecho y documentado en el repositorio?
Se ha comprobado que compila
* [x] ¿He añadido el PR y todos los issues que se resuelvan en él al milestone
correspondiente, que será el primero?
* [x] ¿Cubre totalmente el milestone 0, o hará falta modelización adicional en
el objetivo 4 para poder programar la lógica de negocio?
En principio no haría falta modelización adicional.

Morad02 committed:
Configurando archivo tsconfig.json

Siguiendo las directrices del lenguaje TS

JJ commented on iv.yaml:
```diff
@@ -0,0 +1,4 @@
+lenguaje: TypeScript 
+entidad: 
+  - src/AtletaDisponible.ts
```
Los repositorios nunca deben contener ficheros con capitalización o CamelCase porque dependiendo del sistema de ficheros, pueden no entenderlo. Por favor, acostumbraros a seguir las buenas prácticas no sólo del lenguaje sino también del uso de repos en general.

Morad02 replied:
Nombre de ficheros corregidos teniendo en cuenta las buenas prácticas.

JJ reviewed (changes requested):
Salvo los nombres de ficheros, parece que efectivamente se han creado y seguido una serie de issues, así que puedo estar de acuerdo o no con la solución, pero es la que ha aceptado @MarioGuisado , así que no hay mucho más que añadir.

Morad02 committed:
Corregido el target

Para poder usar los maps se necesita una version posterior a 2015

Morad02 committed:
Cambiando nombre ficheros

Respetando las buenas prácticas

Morad02 committed:
Corregidos errores por cambio de ficheros

Usando las buenas prácticas

Morad02 committed:
Modificado nombre entidades

Siguiendo las buenas prácticas

JJ reviewed (approved):
🥇 A ver, para ser un poco chuminosos, deberías usar camel_case para representar el SnakeCase que se usa en los nombres de los objetos (donde sí es una buena costumbre usarlo). Si puedes hacerlo antes de fusionar, o si no ya que se encargue @MarioGuisado
